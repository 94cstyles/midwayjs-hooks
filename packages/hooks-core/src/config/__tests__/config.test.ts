import { getConfig } from '../'
import { join } from 'path'

test('load user config', () => {
  const fixtures = ['js', 'ts', 'ts-export-default']
  fixtures.forEach((fixture) => {
    expect(loadConfig(fixture)).toMatchSnapshot()
  })
})

function loadConfig(fixture: string) {
  const config = getConfig(join(__dirname, 'fixtures', fixture))
  return config
}
