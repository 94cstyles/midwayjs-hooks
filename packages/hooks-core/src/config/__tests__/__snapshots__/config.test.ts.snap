// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load user config 1`] = `
Object {
  "build": Object {
    "outDir": "./dist",
    "viteOutDir": "./build",
  },
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": true,
}
`;

exports[`load user config 2`] = `
Object {
  "build": Object {
    "outDir": "./dist",
    "viteOutDir": "./build",
  },
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": true,
}
`;

exports[`load user config 3`] = `
Object {
  "build": Object {
    "outDir": "./dist",
    "viteOutDir": "./build",
  },
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": true,
}
`;
